src/sim/TrackData.hh

```diff
@@ -169,7 +169,13 @@ resize(StateData<Ownership::value, M>*                               data,
 
     resize(&data->step_length, size);
     resize(&data->energy_deposition, size);
-    resize(&data->interactions, size);
+
+    // Initialize empty interactions
+    StateCollection<Interaction, Ownership::value, MemSpace::host> interactions;
+    std::vector<Interaction> initial_state(size, Interaction{});
+    make_builder(&interactions)
+        .insert_back(initial_state.begin(), initial_state.end());
+    data->interactions = interactions;
```

Comment:
It would probably be better to instantiate the `Filler` and do the initializations on device rather than copy, or to call a small kernel to clear all the track data as part of `resize`. We can do that in another MR though, I'm too excited about AMD working 😂

Reply:
Haha, sounds good! :)